src/sdk/adapter-registry.js

```diff
@@ -0,0 +1,39 @@
+let instance = null
+
+class AdapterRegistry {
+
+  constructor() {
+    if(!instance){
+      instance = this
+      this.registry = {}
+    }
+    return instance
+  }
+
+  valid(adapter) {
+    if(typeof adapter === 'undefined' || !['get', 'all', 'save', 'filter', 'delete']
```

Comment:
Yes I need because if I don't check the error will be "can't read property 'get' of undefined" and I must check before checking those methods.